include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -900,6 +901,37 @@ namespace internal
 
         const unsigned int end_cell = std::min(mapping_info.cell_type.size(),
                                                std::size_t(cell_range.second));
+
+        // create dummy MappingCollection (TODO: get from somewhere else)
+        dealii::hp::MappingCollection<dim> mapping_collection(mapping);
+
+        // create dummy FECollection (TODO: get from somewhere else)
+        dealii::hp::FECollection<dim> fe_collection;
+        for (unsigned int i = 0; i < n_active_fe_indices; ++i)
+          fe_collection.push_back(dummy_fe);
+
+        // create dummy QCollections (TODO: get from somewhere else)
+        std::vector<dealii::hp::QCollection<dim>> q_collections(
+          mapping_info.cell_data.size());
+
+        for (unsigned int my_q = 0; my_q < mapping_info.cell_data.size();
+             ++my_q)
+          for (const auto &quad : mapping_info.cell_data[my_q].descriptor)
+            q_collections[my_q].push_back(quad.quadrature);
+
+        // fe_values object that is used to compute the mapping data. for
+        // the hp case there might be more than one finite element.
+        std::vector<std::shared_ptr<dealii::hp::FEValues<dim>>> fe_values(
+          mapping_info.cell_data.size());
+
+        for (unsigned int my_q = 0; my_q < mapping_info.cell_data.size();
+             ++my_q)
+          fe_values[my_q] =
+            std::make_shared<dealii::hp::FEValues<dim>>(mapping_collection,
+                                                        fe_collection,
+                                                        q_collections[my_q],
+                                                        update_flags_feval);
```

Comment:
I think it would be appropriate to use a `hp::FEValues` here, since in a nutshell we will be using mapping/fe/quadrature collections in here in the near future. `hp::FEValues` contains the right logic how to select the right `mapping` and `quadrature` (0 or `active_fe_index`) so I don't want to repeat the logic here and at other places.

Setting up the collections (code above) should not be necessary if they are passed here or stored like this at appropriate places. But cleaning up this could be part of a follow-up PR.